File: JavaScript-programs/bubbleSort.js
const arr = [24,10,32,,35,8,17,53,40,15,25];

function bubbleSort(arr) {
  for(let i = 0; i< arr.length - 1; i++) {
    for(let j = 0; j < arr.length - i - 1; j++) {
       if (arr[j] > arr[j+1])
        {
        swap(arr,j,j+1);
        }
    }
  }
}

function swap(arr, prev, next) {
    var temp = arr[prev];
    arr[prev] = arr[next];
    arr[next] = temp;
}

//unsorted array
console.log(arr);

bubbleSort(arr);

//sorted array
console.log(arr);

File: java-programs/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {                                                                      
  public static void main(String []args) {                                              
    int i, j, temp;                                                                
    
    int array[] = {29, 7, 5, 9, 50, 8, 17, 1, 6,11};                                                            
                                                             
    for (i = 0; i < ( array.length - 1 ); i++) {                                                 
      for (j = 0; j < array.length - 1 - i; j++) {                                               
        if (array[j+1] < array[j])      
        {
          temp = array[j];                                                             
          array[j] = array[j+1];                                                         
          array[j+1] = temp;                                                          
        }
      }
    }
    System.out.println("Sorted Array : " + Arrays.toString(array));;;
  } 
}
